File: Serie1/CLI/Serie_1/Serie_1/SyncUtils.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading;
using System.Threading.Tasks;

namespace Serie_1
{
    class SyncUtils
    {
        public static int AdjustTimeout(ref int lastTime, ref int timeout)
        {
            if (timeout != Timeout.Infinite)
            {
                int now = Environment.TickCount;
                int elapsed = (now == lastTime) ? 1 : now - lastTime;
                if (elapsed >= timeout)
                    timeout = 0;
                else
                {
                    timeout -= elapsed;
                    lastTime = now;
                }
            }
            return timeout;
        }
    }
}

File: Serie1/Java/Serie1_PC_1314V/src/Peter/SyncUtils.java
package Peter;

public class SyncUtils {
	public static long adjustTimeout(long lasttime, long timeout) {
	    if (timeout != -1) {
	    	long now = System.currentTimeMillis();
	    	long elapsed = (now == lasttime) ? 1 : (now - lasttime);
	        if (elapsed >= lasttime)
	        {
	            timeout = 0;
	        }
	        else {
	            timeout -= elapsed;
	            lasttime = now;
	        }
	    }
	    return timeout;
	}
}
